crates/scope/src/builder.rs

```diff
@@ -2221,30 +2248,26 @@ impl FunctionScriptStencilBuilder {
         fun.set_has_rest();
     }
 
-    /// Add all closed over bindings of the current function,
-    /// that is calculated by `parameter_scope_builder`.
     fn add_closed_over_bindings(
         &mut self,
-        parameter_scope_builder: &FunctionParametersScopeBuilder,
+        mut closed_over_bindings_for_lazy: Vec<Option<SourceAtomSetIndex>>,
     ) {
-        let closed_over_freevars: HashSet<SourceAtomSetIndex> = parameter_scope_builder
-            .base
-            .name_tracker
-            .closed_over_freevars()
-            .cloned()
-            .collect();
-        let used_freevars: HashSet<SourceAtomSetIndex> = parameter_scope_builder
-            .base
-            .name_tracker
-            .used_freevars()
-            .cloned()
-            .collect();
-        let all_freevars = closed_over_freevars.union(&used_freevars);
+        // Remove trailing `None`s.
+        loop {
+            match closed_over_bindings_for_lazy.last() {
+                Some(Some(_)) => break,
+                None => break,
+                Some(None) => {}
```

Comment:
nit: This code is a bit cryptic, can you briefly comment each case, such as: `binding`, `list is empty` and `end of scope`.